pkg/migrations/op_set_unique.go

```diff
@@ -83,3 +115,48 @@ func addUniqueIndex(ctx context.Context, conn db.DB, table, column, name string)
 
 	return err
 }
+
+func isIndexInProgress(ctx context.Context, conn db.DB, schemaname string, indexname string) (bool, error) {
+	var isInProgress bool
+	rows, err := conn.QueryContext(ctx, `
+		SELECT EXISTS(
+			SELECT * FROM pg_catalog.pg_stat_progress_create_index
+			WHERE index_relid = $1::regclass
+			)`,
+		fmt.Sprintf("%s.%s", pq.QuoteIdentifier(schemaname), pq.QuoteIdentifier(indexname)))
+	if err != nil {
+		return false, fmt.Errorf("getting index in progress with name %q: %w", indexname, err)
+	}
+	if rows == nil {
+		// if rows == nil && err != nil, then it means we have queried a fake db.
```

Comment:
```suggestion
		// if rows == nil && err != nil, then it means we have queried a `FakeDB`.
```

Make it clear we're talking about the `FakeDB` implementation of our `DB` interface.